index.js

```diff
@@ -2,33 +2,67 @@ const Mitm = require('mitm');
 const R = require('ramda');
 const compareHeaders = require('./headers');
 
-const mock = (expectedReq, response) =>
-
+/**
+* @param expectedReqs {object/array} either an array or expected request or a single request
+* @param responses {object/array} either an array of responses or single response object
+* @return promise for the test
+*/
+const mock = (expectedReqs, responses) =>
```

Comment:
I'll have to give this a proper look tomorrow, but simple thing for now, why two separate arrays instead of just one of objects `{ request: { ... }, response: { ... } }`? Then all your code that deals with ensuring the two arrays are of the same size just goes away.

Reply:
It would mean a different API, but that may be a reasonable change

Reply:
Speaking of API design, I'd stop the overloading of one/many and instead split into two functions at the export level:

`mock` that just takes a single request+response combo (use an object for symmetry with the Array one perhaps? ie, js pseudo named args)
and `mockMany` or something like that, which takes the Array.

Reply:
only just saw this after I pushed, I'm not ignoring it intentionally. I'd disagree though, overloading is fairly idiomatic in node and not all that difficult to handle.